packages/react-widgets/src/widgets/LegendWidget.js

```diff
@@ -39,12 +39,22 @@ function LegendWidget({ className, initialCollapsed }) {
     );
   };
 
+  const handleChangeLegendCollapsed = ({ id, collapsed }) => {
+    dispatch(
+      updateLayer({
+        id,
+        layerAttributes: { legend: { collapsed } }
+      })
+    );
+  };
+
   return (
     <LegendWidgetUI
       className={className}
       layers={layers}
       onChangeVisibility={handleChangeVisibility}
       onChangeOpacity={handleChangeOpacity}
+      onChangeLegendCollapsed={handleChangeLegendCollapsed}
```

Comment:
There is already `onCollapsedChange` passed to children, why not reuse it and call dispatch(updateLayer) only as effect response to actually local `collapsed` variable controlled by `setCollapsed` ?